### PR TITLE
feat: Add Locales to Interactions

### PR DESCRIPTION
**Pull request by suneettipirneni**
**Please describe the changes this PR makes and why it should be merged:**

Adds support for locales in interactions.

Ref:
- https://github.com/discord/discord-api-docs/pull/4265

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)

### CODE REVIEW

File: src/structures/Interaction.js [hunk 1]
```diff
@@ -74,6 +74,18 @@ class Interaction extends Base {
      * @type {?Readonly<Permissions>}
      */
     this.memberPermissions = data.member?.permissions ? new Permissions(data.member.permissions).freeze() : null;
+
+    /**
+     * The locale of the member who invoked this interaction
```
Comment (ImRodry) [inline src/structures/Interaction.js:79]:
```suggestion
     * The locale of the user who invoked this interaction
```

File: src/structures/Interaction.js [hunk 2]
```diff
@@ -74,6 +74,18 @@ class Interaction extends Base {
...
+     * @type {string}
+     */
+    this.locale = data.locale;
+
+    /**
+     * The preferred locale from the guild this interaction was sent from
```
Comment (ImRodry) [inline src/structures/Interaction.js:85]:
```suggestion
     * The preferred locale from the guild this interaction was sent in
```

**Review by ImRodry — CHANGES_REQUESTED** [review]

**Review by kyranet — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]

File: typings/index.d.ts [hunk 3]
```diff
@@ -1231,6 +1231,8 @@ export class Interaction<Cached extends CacheType = CacheType> extends Base {
   public user: User;
   public version: number;
   public memberPermissions: CacheTypeReducer<Cached, Readonly<Permissions>>;
+  public locale: string;
+  public guildLocale: string | null;
```
Comment (Jiralite) [inline typings/index.d.ts:1235]: This can be narrowed with all our guild type guard checks, right?

Comment (suneettipirneni) [reply typings/index.d.ts:1235]: Good point! I'll add that

**Review by SpaceEEC — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

**Comment by Smidul** [comment]
Upstream PR merged 👀

**Comment by suneettipirneni** [comment]
Rebased

**Pull request merged**